exes/mqtt-broadcaster/inc/mqtt_broadcaster.hpp

```diff
@@ -0,0 +1,560 @@
+#pragma once
+
+#include <sparkplug_b/sparkplug_b.pb.h>
+#include <async_mqtt/all.hpp>
+#include <boost/asio.hpp>
+
+#include <tfc/dbus/string_maker.hpp>
+
+#include "config.hpp"
+
+using org::eclipse::tahu::protobuf::Payload;
+using org::eclipse::tahu::protobuf::Payload_Metric;
+
+namespace asio = boost::asio;
+namespace details = tfc::ipc::details;
+
+class network_manager {
+public:
+  auto connect_socket(auto&& socket, auto&& resolved_ip) -> asio::awaitable<void> {
+    co_await asio::async_connect(socket, resolved_ip, asio::use_awaitable);
+  }
+
+  auto set_sni_hostname(auto&& native_handle, std::string broker_address) -> void {
+    if (!SSL_set_tlsext_host_name(native_handle, broker_address.c_str())) {
+      boost::system::error_code error{ static_cast<int>(::ERR_get_error()), boost::asio::error::get_ssl_category() };
+
+      throw boost::system::system_error{ error };
+    }
+  }
+
+  auto async_handshake(auto&& socket) -> asio::awaitable<void> {
+    co_await socket.async_handshake(async_mqtt::tls::stream_base::client, asio::use_awaitable);
+  }
+};
+
+struct signal_data {
+  tfc::ipc_ruler::signal information;
+  tfc::ipc::details::any_recv receiver;
+  std::optional<std::any> current_value;
+};
+
+template <class ipc_client_type, class mqtt_client_type, class config_type, class network_manager_type>
+class mqtt_broadcaster {
+public:
+  static constexpr std::string_view namespace_element = "spBv1.0";
+  static constexpr std::string_view ndata = "NDATA";
+  static constexpr std::string_view nbirth = "NBIRTH";
+  static constexpr std::string_view ndeath = "NDEATH";
+  static constexpr std::string_view ncmd = "NCMD";
+  static constexpr std::string_view rebirth_metric = "Node Control/Rebirth";
```

Comment:
are these variables not available from proto definition?

Reply:
They are not to be found in the proto header file at least. Have not found a way to retrieve them.